src/middlewared/middlewared/plugins/interface/netif_linux/interface.py

```diff
@@ -118,26 +114,18 @@ def __getstate__(self, address_stats=False, media=False, vrrp_config=None):
             'vrrp_config': vrrp_config,
         }
 
-        if media:
-            p = subprocess.run(["ethtool", self.name], stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
-                               encoding="utf-8", errors="ignore")
-            if p.returncode == 0:
-                ethtool = {
-                    k.strip(): v.strip()
-                    for k, v in map(lambda s: s.split(":", 1), [line for line in p.stdout.splitlines() if ":" in line])
-                }
-                if "Speed" in ethtool:
-                    bits = [ethtool["Speed"]]
-                    if "Port" in ethtool:
-                        bits.append(ethtool["Port"])
-                    media_subtype = " ".join(bits)
-
-                    state.update({
-                        "media_type": "Ethernet",
-                        "media_subtype": "autoselect" if ethtool.get("Auto-negotiation") == "on" else media_subtype,
-                        "active_media_type": "Ethernet",
-                        "active_media_subtype": media_subtype,
-                    })
+        with EthernetHardwareSettings(self.name) as dev:
+            state.update({
+                'capabilities': dev.enabled_capabilities,
+                'supported_media': dev.supported_media,
```

Comment:
Do we need this if `media` is `False`?

And we'll still call `__mediainfo__` if `media` is `False`

Reply:
Seems we always populate `capabilities` attribute on CORE so I tried to keep the same behavior. However, I thought getting the media info was cheap enough to always get it but I changed it anyways. Or are you saying that we shouldn't populate `capabilities` or any of the media attributes unles `medi == True`?

Reply:
Well, it's really cheap enough, we can drop `media` argument then.